lib/services/local/generic/generic.go

```diff
@@ -173,7 +173,16 @@ func (s *Service[T]) ListResources(ctx context.Context, pageSize int, pageToken
 
 	var nextKey string
 	if len(out) > pageSize {
-		nextKey = backend.GetPaginationKey(out[len(out)-1])
+		lastElementIndex := len(out) - 1
+
+		// Add in any subkeys if present.
+		itemKeyWithoutPrefix := string(result.Items[lastElementIndex].Key[len(s.backendPrefix)+2:])
+		lastSeparatorIndex := strings.LastIndex(itemKeyWithoutPrefix, string(backend.Separator))
+
+		nextKey = backend.GetPaginationKey(out[lastElementIndex])
+		if lastSeparatorIndex != -1 {
+			nextKey = strings.Join(append([]string{}, itemKeyWithoutPrefix[:lastSeparatorIndex], nextKey), string(backend.Separator))
+		}
```

Comment:
Why isn't this basically just `result.Items[len(out)-1].Key[2+len(s.backendPrefix):]`? I'd be wary of changing things like this, since `GetPaginationKey` handles resources with a HostID in a special way.